### PR TITLE
Add Calendar methods: sonarr & readarr

### PR DESCRIPTION
**Pull request by davidnewhall**
Closes half of #66. I'm only going to do Sonarr and Readarr in this 1 PR to keep it smaller. Will make another one for the other two apps.

- [x] Sonarr
- [x] Readarr

Cleans up `Book`, `Author`, and `Episode` types. Probably easier to look at without white-space changes: https://github.com/golift/starr/pull/71/files?w=1

### CODE REVIEW

File: readarr/book.go [hunk 1]
```diff
@@ -14,53 +14,31 @@ import (
...
+	Grabbed        bool           `json:"grabbed"`
+	Overview       string         `json:"overview"`
+	PageCount      int            `json:"pageCount"`
+	Ratings        *starr.Ratings `json:"ratings"`
+	ReleaseDate    time.Time      `json:"releaseDate"`
 	RemoteCover    string         `json:"remoteCover,omitempty"`
-}
-
-// BookAuthor of a Book.
-type BookAuthor struct {
```
Comment (davidnewhall) [inline readarr/book.go:43]: This was removed and replaced by `Book`.

File: readarr/book.go [hunk 2]
```diff
@@ -123,35 +101,13 @@ type AddAuthorOptions struct {
...
-	TitleSlug        interface{}    `json:"titleSlug"`        // Slugs are dumb
+	TitleSlug        string         `json:"titleSlug"`        // Slugs are dumb
 	Images           []*starr.Image `json:"images"`           // this is dumb too
 	ForeignEditionID string         `json:"foreignEditionId"` // GRID ID
 	Monitored        bool           `json:"monitored"`        // true
 	ManualAdd        bool           `json:"manualAdd"`        // true
 }
 
-// AddBookOutput is returned when a book is added.
-type AddBookOutput struct {
```
Comment (davidnewhall) [inline readarr/book.go:134]: This was removed and replaced by `Book`.

File: sonarr/episode.go [hunk 3]
```diff
@@ -16,19 +16,22 @@ const bpEpisode = APIver + "/episode"
...
+	EpisodeNumber            int64          `json:"episodeNumber"`
+	AirDateUtc               time.Time      `json:"airDateUtc"`
+	AirDate                  string         `json:"airDate"`
+	Title                    string         `json:"title"`
+	Overview                 string         `json:"overview"`
+	UnverifiedSceneNumbering bool           `json:"unverifiedSceneNumbering"`
+	HasFile                  bool           `json:"hasFile"`
+	Monitored                bool           `json:"monitored"`
+	Images                   []*starr.Image `json:"images"`
+	Series                   *Series        `json:"series"`
```
Comment (davidnewhall) [inline sonarr/episode.go:34]: I'm not sure how this was ever missed... maybe it was added later?

**Review by Fuochi — APPROVED** [review]

**Comment by davidnewhall** [comment]
Tested this more and found that the pointers were pointless. Removed them. Merged master. Caught a mistake in the last PR and fixed that. Good to go now.

**Pull request merged**